GOVERNANCE.md

```diff
@@ -0,0 +1,253 @@
+
+### Membership Changes
+
+- All team members MUST remain active contributors.
+  If they are unresponsive for > 3 months they will be automatically removed unless a supermajority of the other members of that team agrees to extend the period.
+- A member may voluntarily step down from a team by following the proposal process in the community repo.
+  These proposals are automatically accepted.
+  Once offboarding is complete and the issue is closed, they become a retired member of that team.
+- New members may be added to a team by a supermajority vote by the existing members of that team.
+- When a team has no members the Org team become responsible for it and may either archive the team or find new members.
+- When a team is archived, its members become retired members of that team.
+
+Dev team membership changes have additional considerations:
+
+- To gain and/or maintain Dev team membership the member MUST be a stated maintainer in at least one technical sub-project.
+- Technical sub-project maintainership is stated in a `MAINTAINERS` file at the root of the source code repository.
+- Stated technical sub-project maintainership may be decided by a Simple Majority of existing members listed in the project's `MAINTAINERS` file.
+
+Architecture team membership changes have additional considerations:
+
+- When an Architecture member leaves the team, their stated maintainership of any technical sub-projects will no longer count toward the required minimum of one Architecture team member.
+  If after leaving, there are no other Architecture team members on those sub-projects, the Architecture team is reponsible for deciding how to fulfill that minimum.
+
+Org team membership changes have additional considerations:
+
+- There should be an odd number of Org team member seats, between 3 and 7.
+  An odd number is to avoid a 50/50 decision split.
+  A minimum of 3 is to prevent a single decision point.
+  The current maximum is a guess to keep decision making efficient, and is subject to change.
+- Members of any other team are eligible to be nominated to fill an open Org team member seat.
+- Org members MUST also remain current members of at least one other non-archived team.
+  If that status changes, they will also lose Org team membership.
+- If an Org member voluntarily steps down, in addition to the process above, within 7 calendar days the Flux dev list MUST be notified <https://lists.cncf.io/g/cncf-flux-dev>.
+  This gives contributors reasonable time to be made aware of the opening.
+- When there is an open Org team member seat, any contributor to a repository in the `fluxcd` GitHub org may nominate a suitable existing member of another team as a replacement.
+  - The nomination period will be three weeks starting the day after an Org team member opening becomes available.
+  - The nomination must be made as a reply to the notification topic in the Flux dev list.
+
+## Licenses
+
+- Apache 2.0 is required for all source code repositories.
```

Comment:
I'm wondering if we should explicitly spell out these rules:
- https://github.com/cncf/foundation/blob/master/charter.md#11-ip-policy
- https://github.com/cncf/foundation/blob/master/allowed-third-party-license-policy.md#approved-licenses-for-allowlist
- https://github.com/cncf/foundation/blob/master/copyright-notices.md#copyright-notices

Even if Flux inherits them through being a CNCF project?

Reply:
I believe this is implied as a CNCF project, and other graduated governance docs don't include it either. However that said, I think this could be noted in a single sentence if we want to include it in this section that we intend to comply with these CNCF policies.

Reply:
Never mind I like how you added it in your PR